web/js/codeworld_shared.js

```diff
@@ -532,13 +532,36 @@ function registerStandardHints(successFunc) {
       found[2] = [];
     }
 
-    const options = found[0].concat(found[1]).concat(found[2]);
+    const mappedTerms = {
+      square: 'rectangle',
+      triangle: 'polygon',
+    };
+
+    for (const [lookedUpTerm, substitute] of Object.entries(mappedTerms)) {
+      if (lookedUpTerm.startsWith(term)) {
+        found[2].push({
+          text: window.codeWorldSymbols[substitute].insertText,
```

Comment:
You should check that window.codeWorldSymbols[substitute] exists.  I can arrange for it not to (for instance, but adding `import Prelude()` to the top of the file), and I believe your code will throw exceptions in that case.